scripts/tasks/src/verify-packaging.ts

```diff
@@ -0,0 +1,58 @@
+import assert from 'node:assert/strict';
+import { spawnSync } from 'node:child_process';
+import { readFileSync } from 'node:fs';
+import path from 'node:path';
+
+export function verifyPackaging() {
+  const cwd = process.cwd();
+  const projectJSON: import('@nx/devkit').ProjectConfiguration = JSON.parse(
+    readFileSync(path.join(cwd, 'project.json'), 'utf-8'),
+  );
+  const tags = projectJSON.tags ?? [];
+
+  const result = spawnSync('npm', ['pack', '--dry-run']);
+
+  const processedResult = result.output
+    .toString()
+    .replace(/\bnpm notice\b\s+[\d.]+[kB]+\s+/gi, '')
+    .replace(/[ ]+/g, '');
+
+  const isV8package = tags.indexOf('v8') !== -1;
+  const isV9package = tags.indexOf('vNext') !== -1;
+
+  if (isV9package) {
+    assert.doesNotMatch(processedResult, /src\/[.a-z0-9/-_]+/i, `wont ship source code from "/src"`);
+    assert.doesNotMatch(processedResult, /config\/[.a-z0-9/-_]+/i, `wont ship config folder`);
+    assert.doesNotMatch(processedResult, /etc\/[.a-z0-9/-_]+/i, `wont ship etc folder`);
+    assert.doesNotMatch(processedResult, /just\.config\.[jt]s/i, `wont ship configuration files`);
+  }
+
+  if (isV8package) {
+    assert.match(processedResult, /lib\/[.a-z0-9/-_]+(.d.ts)/i, 'ships dts');
+  }
+
+  assert.match(processedResult, /package\.json/, 'ships package.json');
+  assert.match(processedResult, /README\.md/, 'ships README.md');
+  assert.match(processedResult, /CHANGELOG\.md/, 'ships CHANGELOG.md');
```

Comment:
Wouldn't this throw an error if a package is still private and doesn't contain a CHANGELOG file yet?

Reply:
yup. as I mentioned in the PR description this is not rock solid implementation and is used only on few packages (1 v9, 1 v8) to verify that bump to node 18 wont screw `npm publish`